### PR TITLE
feat: improve shipping method handling

### PR DESCRIPTION
**Pull request by nedu64**
[🍰 Improve handling of shippingMethod selection [S]](https://3.basecamp.com/3091851/buckets/26456130/todos/7912820470)

### CODE REVIEW

File: src/templates/checkout/shipping.twig [hunk 1]
```diff
@@ -11,18 +11,23 @@
...
-			{# If there are no available shipping methods, then just redirect to the payment page #}
-			{% redirect(craft.fostercheckout.getPath('checkout') ~ '/payment') %}
+				{% endfor %}
+			{% elseif cart.availableShippingMethodOptions|length == 0 %}
+				{# If requireShippingMethodSelectionAtCheckout is false and there are no available shipping methods, then just redirect to the payment page #}
+				{% redirect(craft.fostercheckout.getPath('checkout') ~ '/payment') %}
+			{% endif %}
+		{% else %}
+			{# If there's 1 or more shipping method available then set it on the cart #}
+			{% if cart.availableShippingMethodOptions|length == 1 or cart.availableShippingMethodOptions|length > 1  %}
```
Comment (sjcallender) [inline src/templates/checkout/shipping.twig:26]: `cart.availableShippingMethodOptions|length >= 1`

File: src/templates/checkout/shipping.twig [hunk 2]
```diff
@@ -11,18 +11,23 @@
...
 	{% if cart is defined %}
-		{# If there's only 1 shipping method available then set it on the cart #}
-		{% if cart.availableShippingMethodOptions|length == 1 %}
-			{% for handle, option in cart.availableShippingMethodOptions %}
-				{% if option.price == 0 %}
-					{% redirect(craft.fostercheckout.getPath('checkout') ~ '/payment') %}
-				{% else %}
+		{% if currentStore.requireShippingMethodSelectionAtCheckout == false %}
+			{# If there's 1 or more shipping method available then set it on the cart #}
+			{% if cart.availableShippingMethodOptions|length == 1 or cart.availableShippingMethodOptions|length > 1  %}
```
Comment (sjcallender) [inline src/templates/checkout/shipping.twig:16]: `cart.availableShippingMethodOptions|length >= 1`

**Review by sjcallender — COMMENTED** [review]
@nedu64 - Please see small edits

File: src/templates/checkout/shipping.twig (hunk 1 above)
Comment (nedu64) [reply src/templates/checkout/shipping.twig:26]: Fixed in https://github.com/FosterCommerce/foster-checkout/pull/45

File: src/templates/checkout/shipping.twig (hunk 2 above)
Comment (nedu64) [reply src/templates/checkout/shipping.twig:16]: Fixed in https://github.com/FosterCommerce/foster-checkout/pull/45

**Pull request merged**